sklearn/utils/validation.py

```diff
@@ -324,21 +324,26 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None, copy=Fal
     if isinstance(accept_sparse, str):
         accept_sparse = [accept_sparse]
 
+    numeric = dtype == "numeric"
+
     if sp.issparse(array):
-        if dtype == "numeric":
+        if numeric:
```

Comment:
Oops, sorry, I read the code wrong.

I suggest renaming the variable "numeric" to "is_numeric", to avoid stupid people like me misreading the code.